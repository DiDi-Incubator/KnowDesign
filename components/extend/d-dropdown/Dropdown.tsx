import React from 'react';
import {Dropdown, DropDownProps} from 'antd';

const DDropdown = (props: DropDownProps) => {
  return <Dropdown className={'dantd-dropdown'} {...props}/>
};

export default DDropdown;
