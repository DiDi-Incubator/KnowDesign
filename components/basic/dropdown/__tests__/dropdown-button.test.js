import React from 'react';
import { mount } from 'enzyme';
import Dropdown from '..';
import Menu from '../../menu';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';

describe('DropdownButton', () => {
  mountTest(Dropdown.Button);
  rtlTest(Dropdown.Button);

  it('pass appropriate props to Dropdown', () => {
    const props = {
      align: {
        offset: [10, 20],
      },
      overlay: (
        <Menu>
          <Menu.Item key="1">foo</Menu.Item>
        </Menu>
      ),
      disabled: false,
      trigger: ['hover'],
      visible: true,
      onVisibleChange: () => {},
    };

    const wrapper = mount(<Dropdown.Button {...props} />);
    const dropdownProps = wrapper.find(Dropdown).props();

    Object.keys(props).forEach(key => {
      expect(dropdownProps[key]).toBe(props[key]);
    });
  });

  it("don't pass visible to Dropdown if it's not exits", () => {
    const menu = (
      <Menu>
        <Menu.Item key="1">foo</Menu.Item>
      </Menu>
    );
    const wrapper = mount(<Dropdown.Button overlay={menu} />);
    const dropdownProps = wrapper.find(Dropdown).props();

    expect('visible' in dropdownProps).toBe(false);
  });

  it('should support href like Button', () => {
    const menu = (
      <Menu>
        <Menu.Item key="1">foo</Menu.Item>
      </Menu>
    );
    const wrapper = mount(<Dropdown.Button overlay={menu} href="https://ant.design" />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('have static property for type detecting', () => {
    const menu = (
      <Menu>
        <Menu.Item key="1">foo</Menu.Item>
      </Menu>
    );
    const wrapper = mount(<Dropdown.Button overlay={menu} />);
    expect(wrapper.type().__ANT_BUTTON).toBe(true);
  });

  it('should pass mouseEnterDelay and mouseLeaveDelay to Dropdown', () => {
    const menu = (
      <Menu>
        <Menu.Item key="1">foo</Menu.Item>
      </Menu>
    );
    const wrapper = mount(
      <Dropdown.Button mouseEnterDelay={1} mouseLeaveDelay={2} overlay={menu} />,
    );
    expect(wrapper.find('Dropdown').props().mouseEnterDelay).toBe(1);
    expect(wrapper.find('Dropdown').props().mouseLeaveDelay).toBe(2);
  });

  it('should support overlayClassName and overlayStyle', () => {
    const menu = (
      <Menu>
        <Menu.Item key="1">foo</Menu.Item>
      </Menu>
    );
    const wrapper = mount(
      <Dropdown.Button
        overlayClassName="className"
        overlayStyle={{ color: 'red' }}
        overlay={menu}
        visible
      />,
    );
    expect(wrapper.find('.ant-dropdown').getDOMNode().className).toContain('className');
    expect(wrapper.find('.ant-dropdown').getDOMNode().style.color).toContain('red');
  });
});
