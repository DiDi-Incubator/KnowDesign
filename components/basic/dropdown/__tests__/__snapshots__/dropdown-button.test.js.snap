// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownButton rtl render component should be rendered correctly in RTL direction 1`] = `
LoadedCheerio {
  "0": <div
    class="dcloud-btn-group dcloud-btn-group-rtl dcloud-dropdown-button"
  >
    <button
      class="dcloud-btn dcloud-btn-rtl"
      type="button"
    />
    <button
      class="dcloud-btn dcloud-btn-icon-only dcloud-btn-rtl dcloud-dropdown-trigger dcloud-dropdown-rtl"
      type="button"
    >
      <span
        aria-label="ellipsis"
        class="anticon anticon-ellipsis"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="ellipsis"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
          />
        </svg>
      </span>
    </button>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`DropdownButton should support href like Button 1`] = `
LoadedCheerio {
  "0": <div
    class="dcloud-btn-group dcloud-dropdown-button"
  >
    <a
      class="dcloud-btn"
      href="https://ant.design"
    />
    <button
      class="dcloud-btn dcloud-btn-icon-only dcloud-dropdown-trigger"
      type="button"
    >
      <span
        aria-label="ellipsis"
        class="anticon anticon-ellipsis"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="ellipsis"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
          />
        </svg>
      </span>
    </button>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
